clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -361,6 +370,9 @@ private void cancelInFlightRequests(String nodeId,
                 }
             } else if (request.header.apiKey() == ApiKeys.METADATA) {
                 metadataUpdater.handleFailedRequest(now, Optional.empty());
+            } else if ((request.header.apiKey() == ApiKeys.GET_TELEMETRY_SUBSCRIPTIONS ||
+                request.header.apiKey() == ApiKeys.PUSH_TELEMETRY) && telemetrySender != null) {
+                telemetrySender.handleFailedRequest(request.header.apiKey(), null);
```

Comment:
nit: maybe creating an `ApiKeys` from `request.header.apiKey()` would make this visually cleaner?

Reply:
Done.